### ISSUE TITLE
[Bug]: Sorted data table header should have labels for a11y

### ISSUE THREAD

**Issue by rsimpso1** [carbon-design-system/carbon#17768]
### Package

@carbon/web-components

### Browser

Chrome

### Package version

2.13

### React version

18.2.0

### Description

I think the column sorting of the data table breaks some a11y rules. They are buttons but I'm not sure if it would be clear to an SR user what they do, but I'm not sure if that would be considered a bug or not. At a minimum, it seems like if a column is sorted or not is not information available to an SR user. There's no label for the up/down arrows.

### Reproduction/example

https://stackblitz.com/edit/github-kbd9xw-nrubrf?file=index.html

### Steps to reproduce

Sort the columns using the headers and notice that no lables are available to the user on the buttons. (marking as sev 3, I don't have a workaround but it isn't severe.)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

watsonx Assistant

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
